include/opendht/def.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef BUILDING_DLL
+    #ifdef __GNUC__
+      #define RING_PUBLIC __attribute__ ((dllexport))
```

Comment:
Why define with the "RING" keyword? Why not use something that's always relevant to OpenDHT. I suggest changing the macro name to "DHT_PUBLIC" or "OPENDHT_PUBLIC".